interpreter/interpreter.go

```diff
@@ -15,30 +15,31 @@ type Interpreter struct {
 	values  []values.Value
 	options *Scope
 	globals *Scope
-
-	typeSol semantic.TypeSolution
+	types   *TypeScope
 }
 
 // NewInterpreter instantiates a new Flux Interpreter whose builtin values are not mutable.
 // Options are always mutable.
-func NewInterpreter(options, builtins map[string]values.Value) *Interpreter {
+func NewInterpreter(options, builtins map[string]values.Value, types *TypeScope) *Interpreter {
 	optionScope := NewScopeWithValues(options)
 	globalScope := optionScope.NestWithValues(builtins)
 	interpreter := &Interpreter{
 		options: optionScope,
 		globals: globalScope.Nest(),
+		types:   types,
```

Comment:
This line should nest the type scope as it is the non mutable version of NewInterpreter.